src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -18,183 +29,856 @@ object HullmodExoticHandler {
+        return runningFromRefitScreen
+    }
+
+    class Flows {
+        /**
+         * For each child module of [fleetMember] or rather, the fleetMember's [ShipVariantAPI] variant,
+         * the method will first call [shouldInstallHullmodExoticToVariant] followed by [installHullmodExoticToVariant],
+         * with their respective callbacks in proper places
+         */
+        companion object {
+            @JvmStatic
+            fun CheckAndInstallOnAllChildModulesVariants(
+                    fleetMember: FleetMemberAPI,
+                    fleetMemberVariant: ShipVariantAPI,
+                    hullmodExotic: HullmodExotic,
+                    onShouldCallback: OnShouldCallback,
+                    onInstallToChildModuleCallback: OnInstallToChildModuleCallback
+            ) {
+                // First things first, figure out whether we're running from Refit or Exotica screen
+                val isFromRefitScreen = runningFromRefitScreen()
+                val workMode = if (isFromRefitScreen) {
+                    HullmodExoticHandlerWorkMode.LENIENT
+                } else {
+                    HullmodExoticHandlerWorkMode.STRICT
+                }
+
+                // Carry on
+                val moduleSlotList = fleetMemberVariant.moduleSlots
+                logger.info("onInstall()\tmoduleSlots: ${moduleSlotList}")
+                val moduleSlotsNullOrEmpty = moduleSlotList == null || moduleSlotList.isEmpty()
+                if (moduleSlotsNullOrEmpty.not()) {
+                    // Print out the module slot list, and generate a small map of
+                    // <this exotic>, <parent FleetMember> + <list of expected variants> + <list of variants we installed on>
+                    val mutableVariantList = moduleSlotList
+                            .map { slot -> fleetMemberVariant.getModuleVariant(slot) }
+                            .toMutableList()
+                    // Obviously, add the 'member' variant to the list as well
+                    mutableVariantList.add(fleetMemberVariant)
+                    // Now forget about the mutable version and use a immutable version
+                    val variantList = mutableVariantList.toList()
+                    logger.info("onInstall()\tvariantList: ${variantList}")
+
+                    // Iterate through each slot, getting their variant and trying to install there
+                    for (slot in moduleSlotList) {
+                        val moduleVariant = fleetMemberVariant.getModuleVariant(slot)
+                        logger.info("onInstall()\tslot: ${slot}\tmoduleVariant: ${moduleVariant}")
+                        if (moduleVariant == null) continue
+                        val mods = ShipModLoader.get(fleetMember, moduleVariant)
+                        logger.info("onInstall()\tmods: ${mods}")
+                        mods?.let { nonNullMods ->
```

Comment:
somehow, i believe a big chunk of this can just be replaced with 
`getChildModuleVariantList()` extension method we already have in the project.

Sure, we'd have to end up adding the root variant to the variant list again, but the whole moduleSlot and variant lookup should just reuse that extension method.

Reply:
was refactored. `moduleSlots` no longer appear anywhere in the file. resolving.